src/unlimited.cpp

```diff
@@ -1814,8 +1829,22 @@ UniValue getstat(const UniValue &params, bool fHelp)
         }
         else
         {
+            //Get time before request
+            int64_t SeriesStartTime = GetTimeMillis();
             UniValue series = base->GetSeries(seriesStr, count);
+            int64_t SeriesEndTime = GetTimeMillis();
```

Comment:
CStatHistory::timeout is where the samples are combined and promoted to more granular time frame.  That is where you could grab the current time.

You probably need to store it per-time granularity...